### DISCUSSION TITLE
[@types/node] `Blob` type is incompatible with typescripts definition of `Blob` in lib.dom.ts

### DISCUSSION THREAD

**Discussion by jeswr** [DefinitelyTyped/DefinitelyTyped#65049] [Issues with a @types package] [answered]
The `Blob` type defined here https://github.com/microsoft/TypeScript/blob/e83d61398ea0e4231e882121dd6c6bcfe4fdc9e4/src/lib/dom.generated.d.ts#L3075 is not assignable to the `Blob` type in @types/node here https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ec8923e261d3619aef6a4663d50159e7e0dcf3e9/types/node/buffer.d.ts#L121 due to type applied to the `prototype`.

## Sample Code

```ts
import { Blob as BufferBlob } from "buffer";
const bl: Blob = new BufferBlob([]);
```

Causes the typescript compilation error

```
Property 'prototype' is missing in type 'import("buffer").Blob' but required in type 'Blob'.ts(2741)
lib.dom.d.ts(2591, 5): 'prototype' is declared here.
```

I originally raised this [here](https://github.com/microsoft/TypeScript/issues/53668) with the Typescript team; but @RyanCavanaugh has indicated this is a problem with the node type definitions.

[a comment by an automated account is not shown]

**Comment by ExE-Boss** [comment, marked as answer]
This is a bug in **TypeScript** with what happens when `var` and `class` types conflict:
- <https://github.com/microsoft/TypeScript/issues/52166>
